torchbenchmark/models/torchrec_dlrm/__init__.py

```diff
@@ -0,0 +1,156 @@
+                dense_device=device,
+            )
+        elif args.interaction_type == InteractionType.DCN:
+            dlrm_model = DLRM_DCN(
+                embedding_bag_collection=EmbeddingBagCollection(
+                    tables=eb_configs, device=self.device
+                ),
+                dense_in_features=len(DEFAULT_INT_NAMES),
+                dense_arch_layer_sizes=args.dense_arch_layer_sizes,
+                over_arch_layer_sizes=args.over_arch_layer_sizes,
+                dcn_num_layers=args.dcn_num_layers,
+                dcn_low_rank_dim=args.dcn_low_rank_dim,
+                dense_device=device,
+            )
+        elif args.interaction_type == InteractionType.PROJECTION:
+            dlrm_model = DLRM_Projection(
+                embedding_bag_collection=EmbeddingBagCollection(
+                    tables=eb_configs, device=self.device
+                ),
+                dense_in_features=len(DEFAULT_INT_NAMES),
+                dense_arch_layer_sizes=args.dense_arch_layer_sizes,
+                over_arch_layer_sizes=args.over_arch_layer_sizes,
+                interaction_branch1_layer_sizes=args.interaction_branch1_layer_sizes,
+                interaction_branch2_layer_sizes=args.interaction_branch2_layer_sizes,
+                dense_device=device,
+            )
+        else:
+            raise ValueError(
+                "Unknown interaction option set. Should be original, dcn, or projection."
+            )
+        train_model = DLRMTrain(dlrm_model)
+        embedding_optimizer = torch.optim.Adagrad if args.adagrad else torch.optim.SGD
+        # This will apply the Adagrad optimizer in the backward pass for the embeddings (sparse_arch). This means that
+        # the optimizer update will be applied in the backward pass, in this case through a fused op.
+        # TorchRec will use the FBGEMM implementation of EXACT_ADAGRAD. For GPU devices, a fused CUDA kernel is invoked. For CPU, FBGEMM_GPU invokes CPU kernels
+        # https://github.com/pytorch/FBGEMM/blob/2cb8b0dff3e67f9a009c4299defbd6b99cc12b8f/fbgemm_gpu/fbgemm_gpu/split_table_batched_embeddings_ops.py#L676-L678
+        apply_optimizer_in_backward(
+            embedding_optimizer,
+            train_model.model.sparse_arch.parameters(),
+            {"lr": args.learning_rate},
+        )
+        def optimizer_with_params():
+            if args.adagrad:
+                return lambda params: torch.optim.Adagrad(params, lr=args.learning_rate)
+            else:
+                return lambda params: torch.optim.SGD(params, lr=args.learning_rate)
+
+        self.model = train_model
+        dense_optimizer = KeyedOptimizerWrapper(
+            dict(in_backward_optimizer_filter(self.model.named_parameters())),
+            optimizer_with_params(),
+        )
+        # opt = CombinedOptimizer([self.model.fused_optimizer, dense_optimizer])
+        # Only run dense optimizer
+        opt = CombinedOptimizer([dense_optimizer])
```

Comment:
Originally it was `[model.fused_optimizer, dense_optimizer]`. We don't have DDP model here, so removed `fused_optimizer`. Is there a better solution here?